pkg/manifests/manifests.go

```diff
@@ -1491,7 +1491,7 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, telemetrySecret *v1.Secret)
 	return p, nil
 }
 
-func (f *Factory) setupGoGC(p *monv1.Prometheus) {
+func (f *Factory) adjustGoGCConfig(p *monv1.Prometheus) {
```

Comment:
Maybe something like:

```suggestion
func (f *Factory) adjustGoSettings(p *monv1.Prometheus) {
```
Since this affects the GOMEMLIMIT too now.

Reply:
renaming as `adjustGoGCRelatedConfig` because `GOMEMLIMIT` is also related to Garbage collection.